flask-dashboard/dashboard/routings/setup.py

```diff
@@ -0,0 +1,63 @@
+from flask import session, request
+
+from dashboard import user_app, link, version, env, user_var, database_name
+from dashboard.database.endpoint import get_monitor_rule, update_monitor_rule, get_last_accessed_times
+from dashboard.database.monitor_rules import reset_monitor_endpoints
+from dashboard.database.settings import get_setting, set_setting
+from dashboard.forms import MonitorDashboard, ChangeSetting
+from dashboard.measurement import track_performance
+from dashboard.security import secure
+
+
+@user_app.route('/' + link + '/settings', methods=['GET', 'POST'])
+@secure
+def dashboard_settings():
+    form = ChangeSetting()
+    form.username.data = get_setting('username', 'admin')
+    old_password = get_setting('password', 'admin')
+    old_password = 'x' * len(old_password)
+
+    if request.method == 'POST' and form.validate():
+        set_setting('username', form.username.data)
+        if form.password.data:
+            set_setting('password', form.password.data)
+
+    return env.get_template('settings.html').\
+        render(link=link, session=session, version=version, database_name=database_name, user_var=user_var,
+               form=form, old_password=old_password)
+
+
+@user_app.route('/' + link + '/rules', methods=['GET', 'POST'])
```

Comment:
i think the code would be cleaner if, instead of adding the new routes to the app, you would create a new blueprint, and install that blueprint in the app. 

see: http://flask.pocoo.org/docs/0.12/blueprints/